augur/utils.py

```diff
@@ -536,11 +537,15 @@ def load_mask_sites(mask_file):
 
 
 def read_strains(*files, comment_char="#"):
-    """Reads strain names from one or more plain text files and returns the
-    set of distinct strains.
+    print_err("WARNING: augur.utils is not stable. Please use augur.io.read_strains instead of augur.utils.read_strains.")
```

Comment:
Can we add a path to removing this function? E.g. add "this function will be removed in an upcoming major release" or some such?

Reply:
I noticed it is used in [seasonal-flu](https://github.com/nextstrain/seasonal-flu/blob/8519f6521b4c5e958ec7258cabfd1f43bc2944f0/scripts/calculate_epiweek.py#L4) which is why I added the warning instead of removing it.

A clear path sounds like a good idea. I'll update the warning to:

> WARNING: augur.utils is not stable. augur.utils.read_strains will be removed in the future. Please use augur.io.read_strains instead.

We could let it, along with [other planned deprecations](https://github.com/nextstrain/augur/issues/1011), sit and accumulate for a while, then remove them in bulk.

Reply:
That'd be great. Something like a "deprecated.md" file listing function names and a time-to-removal (6 months from when they're added? 3 months?). Then when we make major releases we can check that file and remove the functions?

Reply:
How's 3366d8964585f41955c9b69cd819852923ef0a69...4462ac36614417e8304c4010357df2cdc5f8963c?

I'm not sure what's a good time to removal for `augur.utils.read_strains`. We can update our own references immediately after this PR's changes are released. Maybe just 3 months then?

Reply:
Yeah seems good, but I'd add a timeframe to each one. E.g. "Deprecated December 2023, Planned for removal March 2023 or after"

Reply:
35eb0e25c81cdad762804b8b053aeb8a6eac7b0d